router.go

```diff
@@ -57,52 +58,102 @@ func Convert(ImgPath string, ExhaustPath string, AllowedTypes []string, QUALITY
 			return
 		}
 
-		// Check the original image for existence,
-		if !ImageExists(RawImageAbs) {
-			msg := "Image not found!"
-			c.Send(msg)
-			log.Warn(msg)
-			c.SendStatus(404)
-			return
-		}
-
-		_, WebpAbsPath := GenWebpAbs(RawImageAbs, ExhaustPath, ImgFilename, reqURI)
+		// Start Proxy Mode
+		if proxyMode {
+			// https://test.webp.sh/node.png
+			realRemoteAddr := ImgPath + reqURI
+			// Ping Remote for status code and etag info
 
-		if ImageExists(WebpAbsPath) {
-			finalFile = WebpAbsPath
+			// If status code is 200
+			//     Check for local /node.png-etag-<etagValue>
+			//         if exist
+			//             Send local cache
+			//         else
+			//             Delete local /node.png*
+			//             Fetch and convert to /node.png-etag-<etagValue>
+			//             Send local cache
+			// else status code is 404
+			//      Delete /node.png*
+			//      Send 404
+			fmt.Println("Remote Addr is " + realRemoteAddr + ", fetching..")
+			statusCode, etagValue := GetRemoteImageInfo(realRemoteAddr)
+			if statusCode == 200 {
+				// Check local path: /node.png-etag-<etagValue>
+				localEtagImagePath := ExhaustPath + reqURI + "-etag-" + etagValue
+				if ImageExists(localEtagImagePath) {
+					c.SendFile(localEtagImagePath)
+				} else {
+					// Temporary store of remote file.
+					// ./remote-raw/node.png
+					CleanProxyCache(ExhaustPath + reqURI + "*")
+					localRemoteTmpPath := "./remote-raw" + reqURI
+					FetchRemoteImage(localRemoteTmpPath, realRemoteAddr)
+					q, _ := strconv.ParseFloat(QUALITY, 32)
+					_ = os.MkdirAll(path.Dir(localEtagImagePath), 0755)
+					err := WebpEncoder(localRemoteTmpPath, localEtagImagePath, float32(q), true, nil)
+					if err != nil {
+						fmt.Println(err)
+					}
+					c.SendFile(localEtagImagePath)
+				}
+			} else {
+				msg := "Remote returned + " + strconv.Itoa(statusCode) + " status code!"
```

Comment:
You could just use `fmt.Sprintf("xxx %s xxx")`, it's more efficient than `strconv.Itoa` and concat.